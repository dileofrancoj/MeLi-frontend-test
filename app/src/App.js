import React, { useState } from "react";
import { Route, Switch, Redirect } from "react-router-dom";

import Products from "./Pages/Products";
import Product from "./Pages/Product";
import { Container } from "react-bootstrap";
import Header from "./Components/Header";

export default function App() {
  const [searchItem, setSearchItem] = useState("");
  const handleSearch = (product) => {
    setSearchItem(product.toLowerCase());
  };
  return (
    <React.Fragment>
      <Header handleSearch={handleSearch} />
      <Container>
        <Switch>
          <Route path="/products" exact>
            <Products searchItem={searchItem} />
          </Route>
          <Route path="/product/:id" exact component={Product} />
          <Redirect to="/products" />
        </Switch>
      </Container>
    </React.Fragment>
  );
}
