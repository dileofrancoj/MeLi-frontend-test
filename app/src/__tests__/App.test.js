import React from "react";
import { shallow } from "enzyme";
import { StaticRouter } from "react-router-dom";
import App from "../App";
import { render, screen } from "@testing-library/react";

describe("App", () => {
  test("should render App component for route /", () => {
    const component = shallow(
      <StaticRouter location={"/"}>
        <App />
      </StaticRouter>
    );
    expect(component).toMatchSnapshot();
  });
  test("should render App component for route /items", () => {
    const component = shallow(
      <StaticRouter location={"/items"}>
        <App />
      </StaticRouter>
    );
    expect(component).toMatchSnapshot();
  });
  test("should render App component for route /items/MLA867086863", () => {
    const productId = "MLA867086863";
    const component = shallow(
      <StaticRouter location={`/items/${productId}`}>
        <App />
      </StaticRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
