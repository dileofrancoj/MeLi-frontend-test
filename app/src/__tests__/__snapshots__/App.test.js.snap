// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App component for route / 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  staticContext={Object {}}
>
  <App />
</Router>
`;

exports[`App should render App component for route /items 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/items",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  staticContext={Object {}}
>
  <App />
</Router>
`;

exports[`App should render App component for route /items/MLA867086863 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/items/MLA867086863",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  staticContext={Object {}}
>
  <App />
</Router>
`;
